### PR TITLE
<variables>: Add new global built-in variables

### LINKED ISSUE #20523: Add new global built-in variables
**What would you like to be added**:
${__dashbord.name} : the dashboard name
${__org.name}: the current organization name
${__from.text}: the from variable in string format
${__to.text}: the to variable in string format

**Why is this needed**:
I need to add a title panel on each dashboard to understand on printed page what is visible.
I would like this: 
Overview : Nile (2019-11-19 13:30:40 to 2019-11-20 13:30:42)
With a Text panel with this:
`${__dashbord.name} : ${__org.name} (${__from.text} to ${__to.text})`

### PR DESCRIPTION
**Pull request by dcastanier**
**What this PR does / why we need it**:
It's an implementation of 'Add new global built-in variables #20523'

**Which issue(s) this PR fixes**:
Fixes #20523

**Special notes for your reviewer**:

### CODE REVIEW

**Review by torkelo — CHANGES_REQUESTED** [review]
Not sure I like the new time range variables. Think we can solve that with new formatting options

example: ${__from:datetime}

**Comment by dcastanier** [comment]
Ok with your proposition, new version available

**Comment by torkelo** [comment]
The ideas discussed here https://github.com/grafana/grafana/issues/15903 is also relevant here, need to think a bit more about this

**Comment by dcastanier** [comment]
Any news about this PR ?

File: public/app/features/templating/variable_srv.ts [hunk 1]
```diff
@@ -53,6 +54,14 @@ export class VariableSrv {
...
       .then(() => {
         this.templateSrv.updateIndex();
+        this.templateSrv.setGlobalVariable('__dashboardName', { value: dashboard.title, text: dashboard.title });
+        if (this.$injector.get) {
+          const contextSrv: ContextSrv = this.$injector.get('contextSrv');
+          this.templateSrv.setGlobalVariable('__orgName', {
+            value: contextSrv.user.orgName,
+            text: contextSrv.user.orgName,
+          });
+        }
```
Comment (torkelo) [inline public/app/features/templating/variable_srv.ts]: you do not need to do this, you can just import contextSrv

**Review by torkelo — CHANGES_REQUESTED** [review]
Think this looks ok, just one minor issue

**Comment by dcastanier** [comment]
Any news about this PR ?

**Comment by dcastanier** [comment]
Minor update are done like requested a month ago. 
Stale tag scary me, someone could review it for merge ?
Thanks

**Comment by torkelo** [comment]
Sorry, it needs a bit of a refactor (not sure how exactly) and have. It had time yet

**Comment by dcastanier** [comment]
Thanks Torkelo, let me know if I can help

**Comment by torkelo** [comment]
not stale!

**Comment by dcastanier** [comment]
Hello Torkel, 
I have a production deployment that need this feature for end of February. 
Do you think it's possible to have a release with this PR ?

**Comment by dprokop** [comment]
@dcastanier had a chance to talk with @torkelo today about this. We decided that for variable naming we would like to use the dot notation as we do with data links for example. So, `__dashboardName` would become `__dashboard.name`, and `__orgName` -> `__org.name`. This PR from @torkelo enables support for dot notation in template variables https://github.com/grafana/grafana/pull/21634

**Comment by dcastanier** [comment]
Thanks for your feedback Dominik, look on the good way for `$__dashboard.name` and `$__org.name`.
For date in text format, you think to keep formatting approach like `${__from:datetime}` or use the dot notation like `$__from.text` or `$__from.format` ?

**Comment by dprokop** [comment]
@dcastanier I think `__from.text` is fine. WDYT @torkelo ?

**Comment by dcastanier** [comment]
I have updated this PR to follow the new dot format.
Sorry, I see that merge from trunk as drain a lot of things, I don't know how to remove it.

**Comment by dcastanier** [comment]
To resolve the merge issue, I just recreate a new PR #20523, so I close this one.

**Pull request closed without merging**